tests/integration/framework/test_runner.py

```diff
@@ -48,6 +55,9 @@ def main(self):
         self.add_options(parser)
         self.options = parser.parse_args()
 
+        self.container = DockerRunner(self.options.platform,
+                self.options.image, self.options.verbose)
+
```

Comment:
this forces all tests to run/exec a container. should leave it up to the individual tests in their `run_test` method, by putting this in a separate method that can be overridden by child classes and then call that method here.

Reply:
Fine

Reply:
I changed it, it still do it in this way but does not start automatically background container. It starts it on the first `docker exec` command, making it optional without having another method.

Reply:
what if I want to change the way the container starts? i.e. test for permission escalation?

Reply:
Any example ? No idea how to perform this kind of test